simulator/syncer/syncer.go

```diff
@@ -87,15 +87,27 @@ func (s *Service) addFunc(obj interface{}) {
 	}
 }
 
-func (s *Service) updateFunc(_, newObj interface{}) {
+func (s *Service) updateFunc(oldObj, newObj interface{}) {
 	ctx := context.Background()
-	unstructObj, ok := newObj.(*unstructured.Unstructured)
+	newUnstructuredObj, ok := newObj.(*unstructured.Unstructured)
 	if !ok {
 		klog.Error("Failed to convert runtime.Object to *unstructured.Unstructured")
 		return
 	}
 
-	err := s.resourceApplierService.Update(ctx, unstructObj)
+	oldUnstructuredObj, ok := oldObj.(*unstructured.Unstructured)
+	if !ok {
+		klog.Error("Failed to convert runtime.Object to *unstructured.Unstructured")
+		return
+	}
+
+	if newUnstructuredObj.GetKind() == "Pod" && !podStatusEqual(oldUnstructuredObj, newUnstructuredObj) {
+		if patchErr := s.resourceApplierService.PatchPodStatus(ctx, newUnstructuredObj); patchErr != nil {
+			klog.Errorf("Failed to patch pod status: %v", patchErr)
+		}
+	}
+
+	err := s.resourceApplierService.Update(ctx, newUnstructuredObj)
```

Comment:
The logic of applying resource of real clusters is encapsulated in resourceapplier.
And this fix should also be applied to oneshotimporter and recorder.
So we should confine this fix to resourceapplier.
We could use `defer` to bypass the filters in updateFunc.